### PR TITLE
Add surface temps artifact

### PR DESCRIPTION
**Pull request by imreddyTeja**
Checklist:
- [x] I created a new folder `$artifact_name`
  - [x] I added a `README.md` in that that folder that
    - [x] describes the data and processing done to it
    - [x] lists the sources of the raw data
    - [x] lists the required citation, licenses
  - [x] If applicable (e.g., for Creative Commons), I added a `LICENSE` file
  - [x] I added the scripts that retrieve, process, and produce the artifact
  - [x] I added the environment used for such scripts (typically, `Project.toml`
        and `Manifest.toml`)
  - [x] I added the `OutputArtifacts.toml` file containing the information
        needed for package developers to add `$artifact_name` to their package
- [x] I uploaded the artifact folder to the Caltech cluster (in
      `/groups/esm/ClimaArtifacts/artifacts/$artifact_name`)
- [x] I added the relevant code to the `Overides.toml` on the Caltech Cluster
      (in `/groups/esm/ClimaArtifacts/artifacts/Overrides.toml`)
- [x] I added a link to the main `README.md` to point to the new artifact

This is an example what the calculated absolute temperature looks like:
absolute_temp

### CODE REVIEW

File: surface_temperatures/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+- **temperature** This variable is defined on (longitude × latitude × time), and it contains the monthly mean Air Surface Temperature Anomaly from 1850 to present. Each point measures the local temperature anomaly. There are missing values in this dataset, and the values range from -21.45 to 27.35 in units of degree C. Points are recorded as missing
+if the coverage diagnostic indicates that the locally available data provides less than a 20% constraint on the anomaly.
+
+## Data Usage and Citation
+
+This data is licensed under [Creative Commons BY-NC 4.0 International](https://creativecommons.org/licenses/by-nc/4.0/) for non-commercial use only. Attribution under CC BY-NC terms should be given to Berkeley Earth, including reference to [www.berkeleyearth.org](www.berkeleyearth.org) when possible.
+
+The Berkeley Earth Land/Ocean Temperature Record (December 17, 2020):
+Rohde, R. A. and Hausfather, Z.: , Earth System Science Data, 2020.
```
Comment (imreddyTeja) [inline surface_temperatures/README.md:38]: I'm not certain this is the correct citation for this dataset.

**Comment by szy21** [comment]
BEST data contains monthly climatology and anomalies. Could you preprocess the data and calculate the absolute temperature in the artifact?

**Comment by imreddyTeja** [comment]
The land mask could also be dropped if it is not needed.

File: surface_temperatures/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+There are 2095 points in the time dimension, and the values are held in the `time` variable. Each time point is a decimal, where the integer part of the decimal refers the year, and the decimal refers to the center of the month the data point corresponds to. For example, 1850.125 refers to February 1850, because .125 * 12 = 1.5, which is the center of February. As of the original creation of this artifact, the data ranges from January 1850 to July 2024.
+
+### Spatial Coverage
+
+The spatial variables are defined on a 1 degree latitude by 1 degree longitude grid. The longitude is in degrees east and ranges from -179.5 to 179.5, and the latitude is in degrees north and ranges from -89.5 to 89.5. The values are held in the `latitude` and `longitude` variables.
+
+### `land_mask`
+
+This variable is defined on (longitude × latitude), and its values range from 0.0 to 1.0. The value of each cell is the fraction of the cell the cell that corresponds
```
Comment (szy21) [inline surface_temperatures/README.md:25]:
```suggestion
This variable is defined on (longitude × latitude), and its values range from 0.0 to 1.0. The value of each cell is the fraction of the cell that corresponds
```

File: surface_temperatures/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+to lan (as opposed to large bodies of water.)
```
Comment (szy21) [inline surface_temperatures/README.md:26]:
```suggestion
to land (as opposed to large bodies of water.)
```

File: surface_temperatures/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+#### `absolute_temperature`
+
+This variable is defined on (longitude × latitude × month_number), and it contains the monthly mean Absolute Air Surface Temperature from 1850 to present in degrees C. This variable is produced during pre-processing by combining the `climatology` and `temperature` variables from the downloaded dataset.
+
+`climatology`, which is removed during preprocessing, contains the Air Surface Temperature Climatology for each month, averaged from Jan 1951 to Dec 1980. For example, the first month refers to the estimate average of surface temperature for all Januaries from 1951 to 1980, the second month refers to the mean of all Februarys, and so on.
```
Comment (szy21) [inline surface_temperatures/README.md:32]:
```suggestion
`climatology`, which is removed during preprocessing, contains the Air Surface Temperature Climatology for each month, averaged from Jan 1951 to Dec 1980. For example, the first month refers to the estimate average of surface temperature for all Januaries from 1951 to 1980, the second month refers to the mean of all Februaries, and so on.
```

File: surface_temperatures/LICENSE
Comment (szy21) [inline surface_temperatures/LICENSE:1]: Do we need to add this file, or is the license section in README good enough?

File: surface_temperatures/create_artifact.jl [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+input_ds = NCDataset(FILE_NAME, "r")
+
+# calculate absolute temperature from climatology and anomaly
+climatology = input_ds["climatology"][:, :, :]
+temperature_anomaly = input_ds["temperature"][:, :, :]
+time = input_ds["time"][:]
+temperature_absolute = similar(temperature_anomaly)
+# loop through each time point, and calculate the absolute temperature at it
+for t = 1:input_ds.dim["time"]
+    month = round(Int, 12 * (time[t] % 1) + sqrt(eps(Float64)))
```
Comment (szy21) [inline surface_temperatures/create_artifact.jl:55]:
```suggestion
    month = ceil(Int, 12 * (time[t] % 1) + sqrt(eps(Float64)))
```
I think `ceil` is safer here, and you may not need to sqrt(eps(FT)) anymore?

**Review by szy21 — APPROVED** [review]
@Sbozzolo I don't think this time format will work with ClimaUtilities? Should we preprocess the time dimension here?

**Pull request merged**